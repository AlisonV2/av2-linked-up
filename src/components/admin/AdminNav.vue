<template>
  <header class="header admin-navbar">
    <div class="container-fluid">
      <nav class="navbar-container">
        <ul class="navbar-menu" :class="{ active: isActive }">
          <li class="nav-link">
            <a class="logout-link"
              ><span class="d-none d-sm-inline mx-3" @click.prevent="logout"
                >Logout</span
              >
              <i class=" fs-3 bi bi-box-arrow-right"></i
            ></a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</template>

<script>
/**
 * @exports AdminNavbar
 * @type {Component}
 * @vue-data {boolean} isActive
 * @vue-event logout
 */
export default {
  name: 'Navbar',
  data() {
    return {
      isActive: false,
    };
  },
  methods: {
    /**
     * @description Dispatch logout store action
     * Redirect to Home page if the current route requires authentication
     * @method logout
     */
    logout() {
      this.$store.dispatch('logout');
      if (this.$route.meta.requiresAuth) {
        this.$router.push({ name: 'Home' });
      }
    },
  },
};
</script>
<style lang="scss" scoped>
.admin-navbar {
  background-color: $dark;
  height: 3.5rem;
  .navbar-container {
    display: flex;
    justify-content: flex-end;
  }
  .nav-item {
    color: $light;
  }
  span {
    color: $light;
  }
  .logout-link {
    display: flex;
    align-items: center;
  }
}
</style>
