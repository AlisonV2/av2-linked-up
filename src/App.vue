<template>
  <AdminNav v-if="isUser"/>
  <AppNavbar v-if="!isUser"/>
   <main>
    <router-view />
  </main>
  <AppFooter />
</template>

<script>
import AppNavbar from '@/components/nav/Navbar';
import AppFooter from '@/components/nav/Footer';
import AdminNav from '@/components/admin/AdminNav';

export default {
  name: 'App',
  components: {
    AppNavbar,
    AppFooter,
    AdminNav
  },
  computed: {
    isUser() {
      if (this.$route.meta.isUser) {
        return true;
      }
      return false;
    }
  }
};
</script>
<style lang="scss">
main {
  min-height: 55vh;
}
</style>
