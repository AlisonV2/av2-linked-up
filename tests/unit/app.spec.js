import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';
import App from '@/App';
import AppNavbar from '@/components/nav/Navbar';
import AppFooter from '@/components/nav/Footer';
import AdminNav from '@/components/admin/AdminNav';
import AppModal from '@/components/app/Modal';
import 'lodash';

const actions = {
  initLogin: jest.fn(),
};

const store = createStore({
  actions,
});

const $router = {
  push: jest.fn(),
};

const config = {
  data() {
    return {
      open: false,
    };
  },
  global: {
    plugins: [store],
    components: {
      AppNavbar,
      AppFooter,
      AdminNav,
      AppModal,
    },
    stubs: ['router-view'],
    mocks: {
      $route: {
        meta: {
          requiresAuth: true,
        },
      },
      $router,
    },
  },
};

/**
 * @module AppTest
 */
describe('App.vue', () => {
  it('ArtistInbox Snapshot', () => {
    const component = shallowMount(App, config);
    expect(component.element).toMatchSnapshot();
  });

  it('Renders child component via routing', () => {
    const component = shallowMount(App, {
      data() {
        return {
          open: false,
        };
      },
      global: {
        plugins: [store],
        components: {
          AppNavbar,
          AppFooter,
          AdminNav,
          AppModal,
        },
        stubs: ['router-view'],
        mocks: {
          $route: {
            meta: {
              requiresAuth: false,
            },
          },
          $router,
          methods: {
            getEgg: jest.fn(),
          },
        },
      },
    });
    App.created.call(component.vm);
    expect(actions.initLogin).toHaveBeenCalled();
  });

  it('Renders child component via routing', () => {
    const component = shallowMount(App, config);
    const adminNav = component.findComponent(AdminNav);
    expect(adminNav.exists()).toBe(true);
  });

  it('Modal is not visible', () => {
    const component = shallowMount(App, config);
    const modal = component.findComponent(AppModal);
    expect(modal).not.toBeVisible;
  });

  it('getEgg is triggered', () => {
    const component = shallowMount(App, config);
    component.get('.easter-egg').trigger('mouseover', {
      ctrlKey: true,
    });
    expect(component.vm.open).toBe(true);
  });
});
