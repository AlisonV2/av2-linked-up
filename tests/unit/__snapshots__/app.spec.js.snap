// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue ArtistInbox Snapshot 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="easter-egg"
  />
  <admin-nav-stub />
  <!--v-if-->
  <main>
    <router-view-stub />
  </main>
  <anonymous-stub
    to="#modals"
  />
  <app-footer-stub />
  
</div>
`;
